src/configurationProvider.ts

```diff
@@ -47,10 +56,26 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                 return config;
             } else if (config.request === "launch") {
                 if (!config.mainClass) {
-                    vscode.window.showErrorMessage("Please specify the mainClass in the launch.json.");
-                    this.log("usageError", "Please specify the mainClass in the launch.json.");
-                    return undefined;
-                } else if (!config.classPaths || !Array.isArray(config.classPaths) || !config.classPaths.length) {
+                    const res = <any[]>(await resolveMainClass());
+                    if (res.length === 0) {
+                        vscode.window.showErrorMessage("No main class exists in the workspace. Please use attach instead.");
+                        return;
+                    }
+                    const prefix = "MainClass|Project: ";
+                    const pickItems = res.map((item) => prefix + item.mainClass + "|" + item.projectName);
+                    const selection = await vscode.window.showInformationMessage("Please choose main class and project.").then(
+                        () => vscode.window.showQuickPick(pickItems));
+                    if (typeof(selection) !== "undefined") {

```

Comment:
Please refer to arduino code to improve the pick items. https://github.com/Microsoft/vscode-arduino/blob/master/src/arduino/boardManager.ts#L88